docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -76,6 +75,37 @@ configuration properties as the Hive connector's Glue setup. See
     connector.name=iceberg
     iceberg.catalog.type=glue
 
+REST catalog
+^^^^^^^^^^^^^^
+
+In order to use the Iceberg REST catalog, ensure to configure the catalog type with
+``iceberg.catalog.type=rest`` and provide further details with the following
+properties:
+
+========================================== ============================================================
+Property Name                                        Description
+========================================== ============================================================
+``iceberg.rest.uri``                       REST server API endpoint URI (required).
+                                           Example: ``http://iceberg-with-rest:8181``
+
+``iceberg.rest.security``                  The type of security to use (default: ``NONE``).
+                                           Example: ``OAUTH2``
+
+``iceberg.rest.oauth2.token``              The Bearer token which will be used for interactions
+                                           with the server.
+                                           Example: ``AbCdEf123456``
+
+``iceberg.rest.oauth2.credential``         The credential to exchange for a token in the OAuth2 client
+                                           credentials flow with the server.
+                                           Example: ``AbCdEf123456``
+========================================== ============================================================
+
+.. code-block:: text
+
+    connector.name=iceberg
+    iceberg.catalog.type=rest
+    iceberg.rest.uri=http://iceberg-with-rest:8181
```

Comment:
"iceberg.rest." prefix isn't quire self-explanatory to me. It helps that it's preceded with `iceberg.catalog.type=rest`, but it can be less clear in a bigger configuration file

what about `iceberg.rest-catalog.` config prefix?

Reply:
or `iceberg.catalog.rest.*` that would follow the `iceberg.catalog.type` pretty well.  Thoughts?

Reply:
> or `iceberg.catalog.rest.*`

That would make sense if we did this for other catalog types, but we don't.
For example, for HMS we piggy-back on Hive configs (eg `hive.metastore.uri`).

@electrum thoughts?

Reply:
@findepi After reviewing the other examples of configs, it appears that `iceberg.metastore.rest.*` would be the most consistent with hms, glue, and the proposed jdbc implementations.

hms and glue use: `hive.metastore.*`
rest and jdbc use: `iceberg.metastore.*`